### ISSUE TITLE
Rule: "lines-around-comment". Option "allowBlockStart" does not treat switch block the same as other blocks

### ISSUE THREAD

**Issue by jrencz** [eslint/eslint#9653]
**Tell us about your environment**

* **ESLint Version:** 4.11.0
* **Node Version:** 9.2
* **npm Version:** 5.5.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
  "rules": {
    "lines-around-comment": ["error", {
      "beforeBlockComment": true,
      "beforeLineComment": true,
      "allowBlockStart": true,
      "allowBlockEnd": false,
      "allowObjectStart": true,
      "allowObjectEnd": false,
      "allowArrayStart": true,
      "allowArrayEnd": false,
    }],
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

I bumped into this issue when I added `padded-blocks: ['error', 'never']` and it enforced no additional line above the comment inside switch block in my project. "padded-blocks" does not affect the case that is reported here. I was able to reproduce the problem with only "lines-around-comment" enabled

```js
var a = function () {
  // Foo  (will not trigger lines-around-comment error)
  // Bar
  
  var b = 1;
}

switch (foo) {
  // Foo (will trigger lines-around-comment error)
  // Bar
  
  case 1: {
    // Foo  (will not trigger lines-around-comment error)
  }
}
```

```bash
eslint myfile.js
```

**What did you expect to happen?**
"lines-around-comment" should not report error for line comment at the beginning of a switch block if "allowBlockStart" option is used and it's set to "true"

**What actually happened? Please include the actual, raw output from ESLint.**
The error is reported. Switch block is treated in a different way than other blocks (function body block and case block)

**Demo**
<a href="https://eslint.org/demo/#eyJ0ZXh0IjoiLyplc2xpbnQgbGluZXMtYXJvdW5kLWNvbW1lbnQ6IFtcImVycm9yXCIsIHsgXCJiZWZvcmVCbG9ja0NvbW1lbnRcIjogdHJ1ZSwgXCJiZWZvcmVMaW5lQ29tbWVudFwiOiB0cnVlLCBcImFsbG93QmxvY2tTdGFydFwiOiB0cnVlLCBcImFsbG93QmxvY2tFbmRcIjogZmFsc2UsIFwiYWxsb3dPYmplY3RTdGFydFwiOiB0cnVlLCBcImFsbG93T2JqZWN0RW5kXCI6IGZhbHNlLCBcImFsbG93QXJyYXlTdGFydFwiOiB0cnVlLCBcImFsbG93QXJyYXlFbmRcIjogZmFsc2UgfV0qL1xuXG52YXIgYSA9IGZ1bmN0aW9uICgpIHtcbiAgLy8gRm9vXG4gIC8vIEJhclxuICBcbiAgdmFyIGIgPSAxO1xufVxuXG5zd2l0Y2ggKGZvbykge1xuICAvLyBGb29cbiAgLy8gQmFyXG4gIFxuICBjYXNlIDE6IHtcbiAgICAvLyBGb29cbiAgfVxufSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6NSwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==" target="_blank"><img src="https://www.google.com/webpagethumbnail?c=41&r=4&s=300&d=https://eslint.org/demo/%23eyJ0ZXh0IjoiLyplc2xpbnQgbGluZXMtYXJvdW5kLWNvbW1lbnQ6IFtcImVycm9yXCIsIHsgXCJiZWZvcmVCbG9ja0NvbW1lbnRcIjogdHJ1ZSwgXCJiZWZvcmVMaW5lQ29tbWVudFwiOiB0cnVlLCBcImFsbG93QmxvY2tTdGFydFwiOiB0cnVlLCBcImFsbG93QmxvY2tFbmRcIjogZmFsc2UsIFwiYWxsb3dPYmplY3RTdGFydFwiOiB0cnVlLCBcImFsbG93T2JqZWN0RW5kXCI6IGZhbHNlLCBcImFsbG93QXJyYXlTdGFydFwiOiB0cnVlLCBcImFsbG93QXJyYXlFbmRcIjogZmFsc2UgfV0qL1xuXG52YXIgYSA9IGZ1bmN0aW9uICgpIHtcbiAgLy8gRm9vXG4gIC8vIEJhclxuICBcbiAgdmFyIGIgPSAxO1xufVxuXG5zd2l0Y2ggKGZvbykge1xuICAvLyBGb29cbiAgLy8gQmFyXG4gIFxuICBjYXNlIDE6IHtcbiAgICAvLyBGb29cbiAgfVxufSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6NSwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ%3D%3D&a=AIYkKU9wnz-Xz5HzNrq3u1jHK95SYAywTg"></a>

Related: https://github.com/eslint/eslint/issues/6526

**Comment by not-an-aardvark** [comment]
I think this is working as intended because the curly braces in `switch` statements are technically not blocks. (This is the same reason why we have a separate `allowClassStart` option.) Maybe we should add options like `allowSwitchStart`/`allowSwitchEnd`.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
